client/resource_group/controller/controller.go

```diff
@@ -272,7 +275,8 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
 		var watchMetaChannel, watchConfigChannel chan []*meta_storagepb.Event
 		if !c.ruConfig.isSingleGroupByKeyspace {
 			// Use WithPrevKV() to get the previous key-value pair when get Delete Event.
-			watchMetaChannel, err = c.provider.Watch(ctx, pd.GroupSettingsPathPrefixBytes, opt.WithRev(metaRevision), opt.WithPrefix(), opt.WithPrevKV())
+			prefix := pd.GroupSettingsPathPrefixBytes(c.keyspaceID)
+			watchMetaChannel, err = c.provider.Watch(ctx, prefix, opt.WithRev(metaRevision), opt.WithPrefix(), opt.WithPrevKV())
```

Comment:
Do we have test to cover this watch?

Reply:
I will add it after https://github.com/tikv/pd/pull/9342?notification_referrer_id=NT_kwDOASoxErQxNjU1OTQ0MjM2MjoxOTU0MjI5MA#discussion_r2102537277